atom/common/api/atom_bindings.cc

```diff
@@ -26,6 +28,12 @@ void Crash() {
   static_cast<DummyClass*>(NULL)->crash = true;
 }
 
+void Hang() {
+  // Temporarily access the current thread.
+  base::ThreadRestrictions::ScopedAllowSingleton allow_singleton;
+  base::WaitableEvent(false, false).Wait();
```

Comment:
my understanding was `unresponsive` event on renderer gets triggered when UI thread hangs. So to simulate it, causing a indefinite wait event on UI thread ?